lib/emoji.rb

```diff
@@ -24,14 +24,23 @@ def all
     @all
   end
 
-  def apple_palette
-    return @apple_palette if defined? @apple_palette
-    data = File.open(apple_palette_file, 'r:UTF-8') { |f| JSON.parse(f.read) }
-    @apple_palette = data.fetch('EmojiDataArray').each_with_object({}) do |group, all|
-      title = group.fetch('CVDataTitle').split('-', 2)[1]
-      all[title] = group.fetch('CVCategoryData').fetch('Data').split(',').map do |raw|
-        TEXT_GLYPHS.include?(raw) ? raw + VARIATION_SELECTOR_16 : raw
+  def palette
+    return @palette if defined? @palette
+
+    data = File.open(raw_data_file, 'r:UTF-8') { |f| JSON.parse(f.read) }
+    @palette = data.each_with_object({}) do |group, all|
+      next unless group.key?('category')
+      if data.any? do |e|
+        e.fetch('char') == "#{group.fetch('char')}#{ZERO_WIDTH_JOINER}#{FEMALE_SYMBOL}#{VARIATION_SELECTOR_16}"
+      end && data.any? do |e|
+        e.fetch('char') == "#{group.fetch('char')}#{ZERO_WIDTH_JOINER}#{MALE_SYMBOL}#{VARIATION_SELECTOR_16}"
```

Comment:
It feels like this check to filter out emojis that have gendered variants could be done as a pre-filtering step before `data.each_with_object`? Or, even better, maybe the `raw_data_file` shouldn't contain duplicates to be filtered out at all?

Reply:
I think it makes sense for `raw_data_file` to be a 1:1 mapping to the Unicode Consortium page where they often list a gender neutral emoji before the gendered ones.

So I think a pre-filtering step would make sense here.